File: fsharp/raindrops/Raindrops.fs
module Raindrops

let factors = [3,"Pling";5,"Plang";7,"Plong"]

let convert (number: int): string = 
    let factorSounds = 
        factors 
        |> List.filter (fun (factor,_) -> number % factor = 0) 
        |> List.map (fun (_,sound) -> sound) 
        |> String.concat ""
    match factorSounds with
    | s when s.Length > 0 -> s
    | _ -> string number

File: go/raindrops/raindrops.go
package raindrops

import "strconv"

type rainsound struct {
	factor int
	sound  string
}

var sounds = []rainsound{
	{3, "Pling"},
	{5, "Plang"},
	{7, "Plong"},
}

// Convert returns rainsounds for a given input value
func Convert(input int) string {
	result := ""
	for _, val := range sounds {
		if input%val.factor == 0 {
			result += val.sound
		}
	}

	if len(result) == 0 {
		return strconv.Itoa(input)
	}

	return result
}
